### ISSUE TITLE
cigraph error reported using guess centerline function

### ISSUE THREAD

**Issue by DOSull** [atsyplenkov/centerline#1]
Thanks for an interesting and potentially useful package!

This code 
```
library(dplyr)
library(sf)
library(centerline)

shapes <- st_read("~/Downloads/shapes.gpkg")

centerlines <-
  cnt_path_guess(shapes)
```
Produces this error 
```
Error in shortest_paths(x, from, to, weights = weights, output = "both",  : 
  At vendor/cigraph/src/graph/iterators.c:828 : Cannot create iterator, invalid vertex ID. Invalid vertex ID
```
when `cnt_path_guess(shapes)` is applied to a whole dataframe. 

But no errors are reported when I iterate over each shape as a single polygon 'slice' of the `sf` dataset like this
```
for (i in seq_len(nrow(shapes))) {
  print(i)
  cnt_path_guess(slice(shapes, i))
}
```
I have tried to isolate individual shapes causing the error, but have not been able to do so reliably. Some shapes in this dataset have holes (e.g. row 89), which I thought might be the problem, but that shape is fine, and the loop above suggests that any one of them is, and that the problem only arises when a collection of shapes is run. I'm sorry I can't narrow the problem down any further. Perhaps you have seen this error reported and can comment on what kinds of shape, or issues with shapes might cause it. The GPKG file is in the attached zip file.

[shapes.gpkg.zip](https://github.com/atsyplenkov/centerline/files/15341780/shapes.gpkg.zip)

**Comment by atsyplenkov** [comment]
That's strange. Maybe related to my last commit https://github.com/atsyplenkov/centerline/commit/069347b315813022d76083a2afcb9db302c2a9fa . I will take a look at this later

**Comment by atsyplenkov** [comment]
Oh, that's interesting. Thanks for noticing. I will think it over.

**Comment by atsyplenkov** [comment]
Hey @DOSull! It's been a while! I finally found some time to tackle the issue. Good news—it's not throwing any errors anymore! The only thing is, it’s currently returning just one centerline (the first one).

What are you hoping to get as the output? Should it return a number of centerlines that matches `nrow(shapes)`?

**Comment by DOSull** [comment]
Glad you have found my input useful in tracking down a problem. 

I am honestly unsure what issue I was looking into when I reported this problem. I did expect to get a LINESTRING per polygon rather than a single LINESTRING, but that may or may not be a reasonable expectation - it's really up to you, I think! 

My set of polygons may be a bit strange for generating skeletons from, so I wouldn't necessarily worry too much about fixing your code to work on my problem!
